### PR TITLE
CLI Hardhat plugin

### PR DESCRIPTION
**Pull request by leomassazza**
- Imported interactive-ui as a hardhat plugin 
- Added core-js
- Removed any reference to synthetix v2
- Moved some functions to core-js

### CODE REVIEW

File: packages/cli/.gitignore [hunk 1]
```diff
@@ -1 +1,112 @@
+# Logs
```
Comment (ajsantander) [inline packages/cli/.gitignore:1]: I'd remove this generic gitignore and replace it with a minimal version that only ignores what actually needs to be ignored.

File: packages/cli/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+## hh-cli (Hardhat CLI)
```
Comment (ajsantander) [inline packages/cli/README.md:1]: This file can be deleted for now.

File: packages/cli/tasks-info.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  TASK_INTERACT_NAME: 'interact',
+  TASK_INTERACT_DESC: 'Spins up an interactive ui to interact with contracts.',
+
+  TASK_NOMINATE_OWNER_NAME: 'nominate-owner',
```
Comment (ajsantander) [inline packages/cli/tasks-info.js:5]: @leomassazza what do you think about removing all unimplemented tasks for now? Then we can focus on the interactive task fully and interpret anything present as functional.

File: packages/cli/tasks/accept_ownership.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/*global task*/
+/*eslint no-undef: "error"*/
+/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "[NAME,DESC]" }]*/
+
+const {
+  TASK_INTERACT_NAME,
```
Comment (ajsantander) [inline packages/cli/tasks/accept_ownership.js:6]: I don't think that every task needs to know about every other task, so just import what's needed?

File: packages/core-js/utils/deploymentHelper.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+function getSource({ deploymentData, contract }) {
+  if (contract) return deploymentData.sources[contract];
+  else return deploymentData.sources;
+}
+
+function getDeploymentFilePath({ artifactsPath, network, instance, jsonName }) {
+  console.log({ artifactsPath, network, instance, jsonName });
+  return path.join(artifactsPath, network, instance, jsonName);
+}
```
Comment (ajsantander) [inline packages/core-js/utils/deploymentHelper.js:19]: nit: linter not picking this up?

File: packages/core-js/utils/getEvents.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+async function getPastEvents({ contract, eventName, provider, fromBlock, toBlock }) {
```
Comment (ajsantander) [inline packages/core-js/utils/getEvents.js:1]: Is the interactive task using this? If not, I'd consider removing the file for now. We can always go back to the original CLI implementation and grab it.

File: packages/cli/.gitignore (hunk 1 above)
Comment (leomassazza) [reply packages/cli/.gitignore:1]: Done

File: packages/core-js/utils/deploymentHelper.js (hunk 5 above)
Comment (leomassazza) [reply packages/core-js/utils/deploymentHelper.js:19]: I think linter doesn't find any warning here :) Good catch

File: packages/core-js/utils/getEvents.js (hunk 6 above)
Comment (leomassazza) [reply packages/core-js/utils/getEvents.js:1]: I'll check and remove if not

File: packages/cli/tasks-info.js (hunk 3 above)
Comment (leomassazza) [reply packages/cli/tasks-info.js:5]: ok

File: packages/cli/README.md (hunk 2 above)
Comment (leomassazza) [reply packages/cli/README.md:1]: Done

File: packages/cli/tasks/accept_ownership.js (hunk 4 above)
Comment (leomassazza) [reply packages/cli/tasks/accept_ownership.js]: I removed not implemented tasks and the imports from the implemented ones

File: packages/core-js/utils/deploymentHelper.js (hunk 5 above)
Comment (leomassazza) [reply packages/core-js/utils/deploymentHelper.js:18]: Fixed

File: packages/core-js/utils/getEvents.js (hunk 6 above)
Comment (leomassazza) [reply packages/core-js/utils/getEvents.js]: Don't needed. Removed.

File: packages/core-js/utils/prettyLog.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+const { green, red, cyan, gray } = require('chalk');
```
[the remark this answers is not in the document]
Comment (leomassazza) [reply packages/core-js/utils/prettyLog.js]: Agree. I moved it as private utils for the cli package

**Review by ajsantander — APPROVED** [review]

**Pull request merged**